ada/ast.py

```diff
@@ -8523,18 +8523,18 @@ def stable_properties_assoc_equation():
                 lookup=LK.recursive,
                 from_node=Self.origin_node,
                 categories=all_categories
-            ).filtermap(
-                # It can only refers to a SubpDecl
-                lambda f: f.cast(T.SubpDecl),
-                lambda f: f.is_a(T.SubpDecl)
+            ).filter(
+                # It can only refer to a SubpDecl or an ExprFunction
+                lambda f: Or(
+                    f.cast(T.SubpDecl).then(
+                        lambda f: i.denotes_the_property_function(f.subp_spec)
+                    ),
+                    f.cast(T.ExprFunction).then(
+                        lambda f: i.denotes_the_property_function(f.subp_spec)
+                    )
```

Comment:
We have a `subp_spec_or_null` property, so you could simplify this code to something like:
```
lambda f: f.is_a(SubpDecl, ExprFunction) & i.denotes_the_property_function(f.cast(BasicDecl).subp_spec_or_null))
```

Reply:
Indeed, much better that way!